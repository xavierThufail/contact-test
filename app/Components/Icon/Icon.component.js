import React from 'react';
import IconIonicons from 'react-native-vector-icons/Ionicons';

import config from './Icon.config';

const Icon = ({name, size, transparent}) => {
  return (
    <IconIonicons
      name={name}
      size={size}
      color={transparent ? 'transparent' : "#1b83e3"}
    />
  )
};

Icon.propTypes = config.propTypes;
Icon.displayName = config.displayName;

export default Icon;
