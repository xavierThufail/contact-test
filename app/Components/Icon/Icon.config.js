import PropTypes from 'prop-types';

const propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  transparent: PropTypes.bool
};

const displayName = 'Icon';

export default {
  propTypes,
  displayName
};
