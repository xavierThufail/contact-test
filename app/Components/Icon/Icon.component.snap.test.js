import React from 'react';
import { render } from '@testing-library/react-native';

import Icon from './Icon.component';

describe('Icon Component snapshot test', () => {
  it('Should render correctly', () => {
    const mockedProps = {
      name: 'ios-menu',
      size: 20,
      transparent: true
    };

    const { toJSON } = render(<Icon {...mockedProps} />);

    expect(toJSON()).toMatchSnapshot()
  })
});
