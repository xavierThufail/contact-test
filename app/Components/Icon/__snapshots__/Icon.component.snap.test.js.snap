// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component snapshot test Should render correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "transparent",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ÔèΩ
</Text>
`;
